# Central Intelligence  Agency(CIA) FactBook Data Insights.

## Introduction

The **World Factbook**, also known as the CIA World Factbook is a reference resource produced by the Central Intelligence Agency(CIA)about the countries of the world.

## Approach

Using  SQL in Jupyter Notebook to analyze data from an SQLite database called `factbook.db`.

## Connecting To SQLite Database

In [20]:
%%capture
"""
symbol is used to load the SQLite library, which allows for SQL 
commands to be executed in jupyter notebook

"""
%reload_ext sql 
%sql sqlite:///factbook.db

Getting an overview of the whole database with information about the tables.

In [22]:
%%sql
SELECT *
FROM sqlite_master
WHERE type='table';

 * sqlite:///factbook.db
Done.


[('table', 'sqlite_sequence', 'sqlite_sequence', 3, 'CREATE TABLE sqlite_sequence(name,seq)'),
 ('table', 'facts', 'facts', 47, 'CREATE TABLE "facts" ("id" INTEGER PRIMARY KEY AUTOINCREMENT NOT NULL, "code" varchar(255) NOT NULL, "name" varchar(255) NOT NULL, "area" integer, "a ... (4 characters truncated) ... land" integer, "area_water" integer, "population" integer, "population_growth" float, "birth_rate" float, "death_rate" float, "migration_rate" float)')]

* It is seen that this database consist of two tables, `sqlite_sequence` and `facts`.
* Let us further inspect the `facts` table of this database.

In [25]:
%%sql
SELECT *
FROM facts
LIMIT 5;

 * sqlite:///factbook.db
Done.


[(1, 'af', 'Afghanistan', 652230, 652230, 0, 32564342, 2.32, 38.57, 13.89, 1.51),
 (2, 'al', 'Albania', 28748, 27398, 1350, 3029278, 0.3, 12.92, 6.58, 3.3),
 (3, 'ag', 'Algeria', 2381741, 2381741, 0, 39542166, 1.84, 23.67, 4.31, 0.92),
 (4, 'an', 'Andorra', 468, 468, 0, 85580, 0.12, 8.13, 6.96, 0.0),
 (5, 'ao', 'Angola', 1246700, 1246700, 0, 19625353, 2.78, 38.78, 11.49, 0.46)]

Description for some of these columns goes thus:

* `name` — the name of the country.
* `area`— the country's total area (both land and water).
* `area_land` — the country's land area in square kilometers.
* `area_water` — the country's water area in square kilometers.
* `population` — the country's population.
* `population_growth`— the country's population growth as a percentage.
* `birth_rate` — the country's birth rate, or the number of births per year per 1,000 people.
* `death_rate` — the country's death rate, or the number of death per year per 1,000 people.


Let us compute some summary statistics with the aim of getting some outlier countries id  if they exist.

In [27]:
%%sql
SELECT MIN(population),
       MAX(population), 
       MIN(population_growth), 
        MAX(population_growth)
FROM facts;

 * sqlite:///factbook.db
Done.


[(0, 7256490011, 0.0, 4.02)]

Surprisingly, there exist a country with a population of `0` and there is another country with population of `7256490011`, that is,above **7.2 bilion**, these are clearly outlier but what really are these countries?
Let us Find out.

In [48]:
%%sql 

SELECT *
FROM facts
WHERE population == (SELECT MIN(population) FROM facts);


 * sqlite:///factbook.db
Done.


[(250, 'ay', 'Antarctica', None, 280000, None, 0, None, None, None, None)]

In [49]:
%%sql 

SELECT *
FROM facts
WHERE population == (SELECT MAX(population) FROM facts);

 * sqlite:///factbook.db
Done.


[(261, 'xx', 'World', None, None, None, 7256490011, 1.08, 18.6, 7.8, None)]

The outliers ealier seen are owes to the fact that a row contains information about the whole world explaining the population of **7.2 billion** and there is another row holding information about the **Antarctica**, which is the only continent on earth without human habitation which explains why the population size is **zero**.